cds_ils/ldap/api.py

```diff
@@ -146,196 +135,254 @@ class LdapUserImporter:
+        return UserProfile(
+            user_id=user_id,
+            _displayname="id_{}".format(user_id),
+            full_name=display_name,
+        )
 
-    def import_remote_account(self, user_id, ldap_user):
+    def create_invenio_remote_account(self, user_id, ldap_user):
         """Return new user entry."""
-        return {
-            "user_id": user_id,
-            "extra_data": {
-                "person_id": ldap_user["employeeID"][0].decode("utf8"),
-                "department": ldap_user["department"][0].decode("utf8"),
-            },
-        }
-
-    def import_user_profile(self, user_id, ldap_user):
-        """Return new user profile."""
-        return {
-            "user_id": user_id,
-            "_displayname": "id_" + str(user_id),
-            "full_name": ldap_user["displayName"][0].decode("utf8"),
-        }
-
-    def import_users(self):
-        """Return location and internal location records."""
-
-        def _commit_user(user_data):
-            """Commit new user in db."""
-            user = User(**self.import_user(user_data))
-            db.session.add(user)
-            db.session.commit()
-            return user.id
-
-        for ldap_user in self.ldap_users:
-            print(
-                "Importing user with person id {}".format(
-                    ldap_user["employeeID"][0].decode("utf8")
-                )
-            )
+        employee_id = ldap_user_get(ldap_user, "employeeID")
+        department = ldap_user_get(ldap_user, "department")
+        return RemoteAccount(
+            client_id=self.client_id,
+            user_id=user_id,
+            extra_data=dict(person_id=employee_id, department=department),
+        )
 
-            user_id = _commit_user(ldap_user)
-            identity = UserIdentity(
-                **self.import_user_identity(user_id, ldap_user)
-            )
-            db.session.add(identity)
+    def import_user(self, ldap_user):
+        """Create Invenio users from LDAP export."""
+        user_id = self.create_invenio_user(ldap_user)
 
-            profile = UserProfile(
-                **self.import_user_profile(user_id, ldap_user)
-            )
-            db.session.add(profile)
-            remote_account = RemoteAccount(
-                client_id=self.client_id,
-                **self.import_remote_account(user_id, ldap_user),
-            )
-            db.session.add(remote_account)
+        identity = self.create_invenio_user_identity(user_id, ldap_user)
+        db.session.add(identity)
 
-        db.session.commit()
+        profile = self.create_invenio_user_profile(user_id, ldap_user)
+        db.session.add(profile)
+
+        remote_account = self.create_invenio_remote_account(user_id, ldap_user)
+        db.session.add(remote_account)
 
 
-def import_ldap_users(ldap_users):
+def _reindex_patrons():
+    """Launch the CLI cmd to re-index all users."""
+    from invenio_base.app import create_cli
+
+    cli = create_cli()
```

Comment:
not necessarily for this PR, because we use it everywhere but this does not log exceptions (I tried with `catch_exceptions`, and I think it is not working - for me indexing invoked with this was failing silently)

Reply:
ok interesting, I trusted that code but it is true that I did not verify it. I need to check it because it is important to re-index.

Reply:
as agreed, I will fix in the next PR, it needs changed in `invenio-app-ils` too